tests/lib/linter/linter.js

```diff
@@ -6739,6 +6762,52 @@ var a = "test2";
         });
     });
 
+    describe("options", () => {
+        it("rules should apply meta.defaultOptions on top of schema defaults", () => {
```

Comment:
Note that this is different from `main`. I asked about the code intent: https://github.com/eslint/eslint/pull/17656#discussion_r1491809729